packages/flutter_tools/lib/src/web/file_generators/js/flutter.js

```diff
@@ -243,14 +243,16 @@ _flutter.loader = null;
      * Loads flutter main entrypoint, specified by `entrypointUrl`, and calls a
      * user-specified `onEntrypointLoaded` callback with an EngineInitializer
      * object when it's done.
+     * Appends the `serviceWorkerVersion` to prevent browsers from using the
+     * cached version when a new build exists.
      *
      * @param {*} options
      * @returns {Promise | undefined} that will eventually resolve with an
      * EngineInitializer, or will be rejected with the error caused by the loader.
      * Returns undefined when an `onEntrypointLoaded` callback is supplied in `options`.
      */
     async loadEntrypoint(options) {
-      const { entrypointUrl = `${baseUri}main.dart.js`, onEntrypointLoaded } =
+      const { entrypointUrl = `${baseUri}main.dart.js?v=${serviceWorkerVersion}`, onEntrypointLoaded } =
```

Comment:
I don't think you can use `serviceWorkerVersion` here (if it works in your machine, it's because it's reading the global variable from window, I'd say!).

Can you please check how the configuration is separated in line ~350 `async loadEntrypoint(options)`, and make sure that your fix passes the *parameterized* serviceWorkerVersion down to this method call (not the global one?)